chirp/drivers/kenwood_d7.py

```diff
@@ -13,332 +13,3404 @@
 
     def get_settings(self):
-        main = RadioSettingGroup("main", "Main")
-        aux = RadioSettingGroup("aux", "Aux")
-        tnc = RadioSettingGroup("tnc", "TNC")
-        save = RadioSettingGroup("save", "Save")
-        display = RadioSettingGroup("display", "Display")
-        dtmf = RadioSettingGroup("dtmf", "DTMF")
-        radio = RadioSettingGroup("radio", "Radio",
-                                  aux, tnc, save, display, dtmf)
-        sky = RadioSettingGroup("sky", "SkyCommand")
-        aprs = RadioSettingGroup("aprs", "APRS")
-
-        top = RadioSettings(main, radio, aprs, sky)
-
-        bools = [("AMR", aprs, "APRS Message Auto-Reply"),
-                 ("AIP", aux, "Advanced Intercept Point"),
-                 ("ARO", aux, "Automatic Repeater Offset"),
-                 ("BCN", aprs, "Beacon"),
-                 ("CH", radio, "Channel Mode Display"),
-                 # ("DIG", aprs, "APRS Digipeater"),
-                 ("DL", main, "Dual"),
-                 ("LK", main, "Lock"),
-                 ("TSP", dtmf, "DTMF Fast Transmission"),
-                 ]
-
-        for setting, group, name in bools:
-            value = self._kenwood_get_bool(setting)
-            rs = RadioSetting(setting, name,
-                              RadioSettingValueBoolean(value))
-            group.append(rs)
+        if self._setting_count == 0:
+            self._count_settings(self._SETTINGS_MENUS)
+        ret = self._create_group(self._SETTINGS_MENUS)
+        self._cur_setting = 0
+        self._refresh_settings()
+        status = chirp_common.Status()
+        status.cur = self._setting_count
+        status.max = self._setting_count
+        status.msg = "%-40s" % "Done"
+        self.status_fn(status)
+        return ret
 
-        lists = [("BEP", aux, "Beep"),
-                 ("BEPT", aprs, "APRS Beep"),
-                 ("DS", tnc, "Data Sense"),
-                 ("DTB", tnc, "Data Band"),
-                 ("DTBA", aprs, "APRS Data Band"),
-                 ("DTX", aprs, "APRS Data TX"),
-                 # ("ICO", aprs, "APRS Icon"),
-                 ("PKSA", aprs, "APRS Packet Speed"),
-                 ("POSC", aprs, "APRS Position Comment"),
-                 ("PT", dtmf, "DTMF Speed"),
-                 ("TEMP", aprs, "APRS Temperature Units"),
-                 ("TXI", aprs, "APRS Transmit Interval"),
-                 # ("UNIT", aprs, "APRS Display Units"),
-                 ("WAY", aprs, "Waypoint Mode"),
-                 ]
-
-        for setting, group, name in lists:
-            value = self._kenwood_get_int(setting)
-            options = self._get_setting_options(setting)
-            rs = RadioSetting(setting, name,
-                              RadioSettingValueList(options,
-                                                    options[value]))
-            group.append(rs)
+    def set_memory(self, memory):
+        index, memid = self._validate_memory(memory)
+        if (memory.empty):
```

Comment:
No need for parens here.

Reply:
Damn, I thought I managed to catch myself on all of these, thanks.